HospitalToday/Services/Implementation/ReportService.cs

```diff
@@ -29,14 +29,32 @@ public void Remove(Report item)
             reportRep.Delete(item.Id);
         }
 
-        public List<Report> GetList()
+        public IList<Report> GetList()
         {
             return reportRep.GetList();
         }
 
         public Report GetItemById(int id)
         {
             return reportRep.GetItem(id);
-        }           
+        }
+
+        public IList<int> GetListId()
+        {
+            return reportRep.GetList().Select(x => x.Id).ToList();
+        }
+
+        public double GetReportTotalCost(int id)
+        {
+            var report = reportRep.GetItem(id);
+            if (report != null && report.Medicines != null)
```

Comment:
Take a look, if you really need check for null for Medicines collection.
Also it will be better to check for Any()

Reply:
Added changes